content/master/concepts/compositions.md

```diff
@@ -1308,47 +1308,74 @@ parameters in the Composition.
 * Composition function `type` is `container`.
 * Composition function names are unique.
 
-### Resource schema validation
-<!-- vale write-good.TooWordy = NO -->
-Optionally, Crossplane can also validate the schema of the resources defined
-inside a Composition. This verifies that the resource `apiVersion` and `kinds`
-are valid. 
-<!-- vale write-good.TooWordy = YES -->
+### Composition schema aware validation
+
+Crossplane also performs schema aware
+validation of Compositions. Schema validation checks that `patches`, 
+`readinessChecks` and `connectionDetails` are valid according to the resource 
+schemas. For example, checking that the source and destination fields of a patch 
+are valid according to the source and destination resource schema.
+
+{{<hint "note" >}}
+Composition schema aware validation is a beta feature. Crossplane enables
+beta features by default. 
+
+Disable schema aware validation by setting the
+`--enable-composition-webhook-schema-validation=false` flag on the Crossplane
+pod.  
 
-Enable "schema validation" with the
-`--enable-composition-webhook-schema-validation` flag on the Crossplane pod.
 The [Crossplane Pods]({{<ref "./pods#edit-the-deployment">}}) page has
 more information on enabling Crossplane flags.
+{{< /hint >}}
+
+#### Schema aware validation modes
+
+Integrity validation errors always result in a rejection of the Composition.
+
+You can configure how Crossplane should handle both missing involved resources'
+schemas and actual schema aware validation errors.
 
 {{<hint "note" >}}
-<!-- vale write-good.TooWordy = NO -->
-Schema validation only checks the `apiVersion` and `kind` are valid. Schema
-validation doesn't validate the fields of a specific resource.
-<!-- vale write-good.TooWordy = YES -->
+In case of any involved resource schema missing, Crossplane skips schema aware
+validation altogether.
```

Comment:
```suggestion
If a resource is missing a schema, Crossplane skips schema aware and integrity 
checking.

Crossplane only provides "missing schema" warnings or errors.
```

Reply:
Nope, just schema aware validation is skipped, not integrity checks. See the full logic below 👇 read "valid per se" as integrity check
 ![image](https://github.com/crossplane/docs/assets/5697904/c25949dd-723a-4556-a451-75dc82bbae93)

Reply:
updated

Reply:
Not sure I get what you mean with the last sentence in the updated version.

The first sentence is an ok rewriting of the original version, although a bit oversimplified. I would be happier keeping the original version tbh

Reply:
The last sentence is to be explicit about what is still being done. It's implied when you say. "skips schema aware" but this adds in meaning of "it still runs the other two checks"

Reply:
`Crossplane only provides "missing schema" and "integrity checking" warnings or errors.` 

The wording and the fact makes it sound like if Crossplane always only provided these two things.

A more correct version would be:
```suggestion
If a resource schema is missing, Crossplane skips schema aware validation altogether, but will still return an error in case of any integrity error and either a warning or an error depending on the mode selected for the missing schemas.
```
But I get Vale would probably not like it and it's a long sentence, which are the reasons why I tried avoiding getting in so much details.

Reply:
I like that. Slight edit for our angry overlord Vale:

> If a resource schema is missing, Crossplane skips schema aware validation 
> but still returns an error for integrity errors and a warning or an error 
> for the missing schemas.